api/agents.go

```diff
@@ -2,18 +2,29 @@ package api
 
 // AgentRegisterRequest is a call to register on the Buildkite Agent API
 type AgentRegisterRequest struct {
-	Name               string   `json:"name"`
-	Hostname           string   `json:"hostname"`
-	OS                 string   `json:"os"`
-	Arch               string   `json:"arch"`
-	ScriptEvalEnabled  bool     `json:"script_eval_enabled"`
-	IgnoreInDispatches bool     `json:"ignore_in_dispatches"`
-	Priority           string   `json:"priority,omitempty"`
-	Version            string   `json:"version"`
-	Build              string   `json:"build"`
-	Tags               []string `json:"meta_data"`
-	PID                int      `json:"pid,omitempty"`
-	MachineID          string   `json:"machine_id,omitempty"`
+	Name               string        `json:"name"`
+	Hostname           string        `json:"hostname"`
+	OS                 string        `json:"os"`
+	Arch               string        `json:"arch"`
+	ScriptEvalEnabled  bool          `json:"script_eval_enabled"`
+	IgnoreInDispatches bool          `json:"ignore_in_dispatches"`
+	Priority           string        `json:"priority,omitempty"`
+	Version            string        `json:"version"`
+	Build              string        `json:"build"`
+	Tags               []string      `json:"meta_data"`
+	PID                int           `json:"pid,omitempty"`
+	MachineID          string        `json:"machine_id,omitempty"`
+	FeatureUsage       *FeatureUsage `json:"feature_usage"`
```

Comment:
Could we call this "telemetry" instead? Feels like a more general concept, and matches the arg name.

Reply:
eh... i think telemetry covers a larger concept that what we have here - ie, if we were sending back job timings to buildkite (for buildkite to see and not the client), that would also be another type of telemetry, that i think we would want to disable that using the same flag that disables this. Does that make sense? I feel like i haven't explained it that well ahah

Reply:
Yeah, true. I wish we didn't already use the term "meta_data" 😬 

Maybe feature_usage works. It just feels odd in the backend.

Reply:
maybe `agent_feature_usage` is a little bit less odd?

Reply:
Maybe the other direction, what about just `features`?

Reply:
I'm overthinking it.

I've adjusted the backend to accept `feature_usage` as a json blob.